benchexec/tools/ultimate.py

```diff
@@ -1,12 +1,12 @@
 # This file is part of BenchExec, a framework for reliable benchmarking:
 # https://github.com/sosy-lab/benchexec
 #
-# SPDX-FileCopyrightText: 2007-2020 Dirk Beyer <https://www.sosy-lab.org>
-# SPDX-FileCopyrightText: 2015 Daniel Dietsch
+# SPDX-FileCopyrightText: 2015-2020 Daniel Dietsch <dietsch@informatik.uni-freiburg.de>
```

Comment:
Please keep the existing copyright statements. The license requires that we "must cause any modified files to carry prominent notices stating that we changed the files", which we do by adding our copyright line. And there have and probably will be changes and contributions from us to these files. Furthermore, one could consider tool-info modules derivative works of BenchExec because they are tightly coupled and based on our API definitions and are typically inspired or have code copied from the template module or other tool-info modules.

Note that such a copyright declaration *does not change* who owns the copyright at all anyway (this is defined by law) and it thus does not diminish any rights that one might have, it is just a (hopefully true) description of facts. It also does not say anything about who owns copyrights to which parts or about a quantitative relationship or whether this is an exhaustive list. In this case it is just the statement that at least you and Dirk claim to hold at least some copyright over some content of this file.

(Although it does not strictly apply in this case, also note that if you redistribute files under the Apache license, you must keep all relevant existing copyright etc. notices, otherwise it is a license violation. So one needs to be really careful when removing such notices.)

Of course, updating the line with your copyright is totally fine.

Reply:
The old copyright statement was added in https://github.com/sosy-lab/benchexec/commit/d0f03ac52be3ba95ace4c249b6d67cf487b18c26#diff-0a9af880936ddc6aa5d95be815747dd617133804c4b2a9bc3e9fc8443d33d0ec, where Dirk Beyer was also added (in particular with a strange timeframe that goes before the inception of the file itself).

We can add any additional contributor, of course. But perhaps some kind of rule when a contribution is significant enough to change the header would be nice to have.

I leave the question whether implementing an API is derivative work to someone else. I believe not ;)

Reply:
> The old copyright statement was added in [d0f03ac#diff-0a9af880936ddc6aa5d95be815747dd617133804c4b2a9bc3e9fc8443d33d0ec](https://github.com/sosy-lab/benchexec/commit/d0f03ac52be3ba95ace4c249b6d67cf487b18c26#diff-0a9af880936ddc6aa5d95be815747dd617133804c4b2a9bc3e9fc8443d33d0ec), where Dirk Beyer was also added (in particular with a strange timeframe that goes before the inception of the file itself).

Yes, when we changed the header to be machine readable we also made sure to update the information in it to be more accurate where applicable (and for example also to handle the license requirement that we must notify about changed files).

The year in the copyright declaration has no legal purpose, it serves just for documentation. For simplicity we decided to just add the same range of years (that covers BenchExec and the origin of its source from before it became BenchExec). Otherwise we would need to investigate whether there is some content in each file that might have been copied from other older sources etc., but this is clearly not worth it.

> But perhaps some kind of rule when a contribution is significant enough to change the header would be nice to have.

This is a question that we would like to avoid in general, because we are no legal experts and cannot answer this (and its answer would change depending on where you are in the world). So if someone changes a file and requires their copyright to be added, we are likely to agree.

> I leave the question whether implementing an API is derivative work to someone else. I believe not ;)

It is irrelevant here, but the Linux kernel developers and the Linux distributions would disagree with you (this is why proprietary Linux kernel modules are not bundled with Linux kernels).

Reply:
Is the new copyright header ok for you?

The question of "derivative work" does indeed not apply to copyright headers. The Linux kernel does not bundle proprietary Linux kernel modules because the licenses are incompatible.

Reply:
No, please don't add other people of which you don't know whether they would like to be listed or not.

(And in fact, in this case it is even wrong, I do not own any copyright on parts of BenchExec, and neither does Karlheinz.)

Reply:
Ok, I can remove you guys and keep only Dirk.

Reply:
But for the record: I do not believe that Dirk has a copyright on any of the files in this commit.

Reply:
Luckily that question does not need to be solved now. But if you think that Karlheinz and me could claim copyright over parts of the code here (as indicated by f3e78d2), then you should also think the same about Dirk, because he owns the copyright for our contributions.

Reply:
Fair enough.